secretsclient/client.go

```diff
@@ -102,6 +114,36 @@ func (c *client) doCall(method, url string, opts ...requestOpt) (int, string, er
 			return 0, "", err
 		}
 	}
+
+	var status int
+	var body string
+	var err error
+	duration := time.Millisecond * 100
+	for attempt := 0; attempt < maxAttempt; attempt++ {
+		status, body, err = c.doCallImp(r, method, url, opts...)
+		if err == nil && status < 500 {
+			return status, body, err
+		}
+		if err != nil {
+			c.warnFunc("retring http request due to %v", err)
+		} else {
+			msg, err := getMessageFromJSON(bytes.NewReader([]byte(body)))
+			if err == nil {
+				c.warnFunc("retring http request due to unexpected status code %v: %v", status, msg)
+			} else {
+				c.warnFunc("retring http request due to unexpected status code %v", status)
+			}
+		}
+		if duration > maxSleepBeforeRetry {
+			duration = maxSleepBeforeRetry
+		}
+		time.Sleep(duration)
+		duration *= 2
+	}
+	return status, body, err
+}
```

Comment:
Wondering if we should use some library for this. Although this is pretty small and simple too.

Reply:
I was half tempted to use one, but figured less (dependencies) is more.